### PR TITLE
Charlesmchen/archive view indicator

### PR DESCRIPTION
**Pull request by charlesmchen**
* Convert home view to programmatic layout.
* Add a reminder that you are in archive view. 
* Fix bug wherein reminder view's hot area was too small.
* Fix bug wherein "current" thread wasn't always visible when you returned to home view.

PTAL @michaelkirk 

screen shot 2017-07-10 at 4 38 24 pm

### CODE REVIEW

File: Signal/src/views/ReminderView.swift [hunk 1]
```diff
@@ -51,10 +51,10 @@ class ReminderView: UIView {
         self.backgroundColor = UIColor.ows_reminderYellow()
         self.clipsToBounds = true
 
-        let container = UIView()
-
         let tapGesture = UITapGestureRecognizer(target: self, action: #selector(handleTap(gestureRecognizer:)))
-        container.addGestureRecognizer(tapGesture)
```
Comment (charlesmchen) [inline Signal/src/views/ReminderView.swift]: The container is significantly smaller than the entire reminder view.

File: Signal/src/network/PushManager.m [hunk 2]
```diff
@@ -155,7 +155,6 @@ - (void)application:(UIApplication *)application
                 success:^{
                     // TODO do we really want to mark them all as read?
                     [self markAllInThreadAsRead:notification.userInfo completionHandler:completionHandler];
-                    [[[[Environment getCurrent] signalsViewController] tableView] reloadData];
```
Comment (charlesmchen) [inline Signal/src/network/PushManager.m]: This shouldn't be necessary; marking messages as read "touches" the messages' threads so the home view will get a "yap database modified" event.

**Comment by charlesmchen** [comment]
PTAL @michaelkirk I'd like to do some RTL work that on top of this PR.

File: Signal/translations/en.lproj/Localizable.strings [hunk 3]
```diff
@@ -643,6 +643,9 @@
 /* Call setup status label */
 "IN_CALL_TERMINATED" = "Call Ended.";
 
+/* Label reminding the user that they are in archive mode. */
+"INBOX_VIEW_ARCHIVE_MODE_REMINDER" = "You're in archive mode.  Tap to return to your Inbox.";
```
Comment (charlesmchen) [inline Signal/translations/en.lproj/Localizable.strings]: Done.

File: Signal/src/ViewControllers/SignalsViewController.m [hunk 4]
```diff
@@ -299,6 +365,30 @@ - (void)viewWillAppear:(BOOL)animated {
 
     self.isViewVisible = YES;
     [self checkIfEmptyView];
+
+    // When returning to home view, try to ensure that the "last" thread is still
+    // visible.  The threads often change ordering while in conversation view due
+    // to incoming & outgoing messages.
+    if (self.lastThread) {
```
Comment (charlesmchen) [inline Signal/src/ViewControllers/SignalsViewController.m]: Better, thanks.

File: Signal/src/ViewControllers/SignalsViewController.m [hunk 5]
```diff
@@ -695,14 +787,18 @@ - (void)presentViewControllerWithBlock:(void (^)())presentationBlock animateDism
     }
 }
 
-#pragma mark - IBAction
+#pragma mark - Groupings
 
-- (IBAction)selectedInbox:(id)sender {
+- (void)selectedInbox
+{
+    self.segmentedControl.selectedSegmentIndex = 0;
```
Comment (charlesmchen) [inline Signal/src/ViewControllers/SignalsViewController.m]: Previously the only way to switch between these "groups" was through the segmented control, so it wasn't necessary to sync it with these changes. Now we can switch from the "reminder" that I've just added.

Comment (michaelkirk) [inline Signal/src/ViewControllers/SignalsViewController.m]: ah right. Makes sense.

**Pull request merged**